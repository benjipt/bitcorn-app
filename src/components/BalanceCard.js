import React from 'react'

export default function BalanceCard({ balance }) {
    return (
        <div className="container text-center mt-5 p-4 border-black rounded card-custom" data-testid="BalanceCard-1">
            <h5>JobCoin Balance</h5>
            <hr></hr>
            <p className="balance">{ balance }</p>
        </div>
    )
}
