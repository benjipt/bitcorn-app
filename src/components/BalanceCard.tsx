interface BalanceCardProps {
  balance: string
}

export default function BalanceCard({ balance }: BalanceCardProps) {
    return (
        <div className="container text-center mt-5 p-4 border-black rounded card-custom" data-testid="BalanceCard-1">
            <h5>JobCoin Balance</h5>
            <hr></hr>
            <p className="balance">{ balance }</p>
        </div>
    )
}
