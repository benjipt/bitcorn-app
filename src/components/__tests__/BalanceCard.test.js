import { render, screen, cleanup } from '@testing-library/react'
import renderer from 'react-test-renderer'
import BalanceCard from '../BalanceCard'

afterEach(() => {
    cleanup()
})

describe('BalanceCard', () => {

    test('should render BalanceCard component', () => {
        render(<BalanceCard/>)
        const BalanceCardComponent = screen.getByTestId('BalanceCard-1')
        expect(BalanceCardComponent).toBeInTheDocument()
    })

    test('passed props should behave as expected', () => {
        const balance = '150'

        render(<BalanceCard balance={ balance } />)
        const BalanceCardComponent = screen.getByTestId('BalanceCard-1')
        expect(BalanceCardComponent).toHaveTextContent(balance)
    })

    test('matches snapshot', () => {
        const balance = '150'

        const tree = renderer.create(<BalanceCard balance={ balance } />).toJSON()
        expect(tree).toMatchSnapshot()
    })
})