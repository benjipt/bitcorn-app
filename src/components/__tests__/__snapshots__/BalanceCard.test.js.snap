// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BalanceCard matches snapshot 1`] = `
<div
  className="container text-center mt-5 p-4 border-black rounded card-custom"
  data-testid="BalanceCard-1"
>
  <h5>
    JobCoin Balance
  </h5>
  <hr />
  <p
    className="balance"
  >
    150
  </p>
</div>
`;
